internal/features/features.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package features sets the feature gates that
+// source-controller supports, and their default
+// states.
+package features
+
+import feathelper "github.com/fluxcd/pkg/runtime/features"
+
+const (
+	// OptimizedGitClones decreases resource utilization for GitRepository
+	// reconciliations. It supports both go-git and libgit2 implementations
+	// when cloning repositories using branches or tags.
+	//
+	// When enabled, avoids full clone operations by first checking whether
+	// the last revision is still the same at the target repository,
+	// and if that is so, skips the reconciliation.
+	GitOptimizedClones = "git-optimized-clones"
```

Comment:
maybe?
```suggestion
	OptimizedGitClones = "git-optimized-clones"
```

Reply:
I would continue to prefix it as this provides categorization for free (e.g. `GitFeatureA`, `GitFeatureB`, etc.). Which is easier with e.g. auto suggestions.

Reply:
@pjbgf and I discussed this, and we landed on the conclusion that the const name should be `OptimizedGitClones` since that sounds better but the flag will be prefixed with `git-` to get categorization for free like you said.

Reply:
But that does not help me while working in an IDE, which is what I care about most while making this suggestion.